service/matching/taskReader.go

```diff
@@ -421,3 +376,120 @@ func (tr *taskReader) newDispatchContext(isolationGroup string) (context.Context
 	}
 	return tr.cancelCtx, func() {}
 }
+
+func (tr *taskReader) dispatchSingleTaskFromBufferWithRetries(isolationGroup string, taskInfo *persistence.TaskInfo) (breakDispatchLoop bool) {
+	// retry loop for dispatching a single task
+	for {
+		breakDispatchLoop, breakRetryLoop := tr.dispatchSingleTaskFromBuffer(isolationGroup, taskInfo)
```

Comment:
hmmmmm.  double boolean :|
usage looks correct, just wondering if this should be something else to reduce risk...

a brief stab at enums feels even worse, but what if it was a result struct?
```go
// return
return dispatchResult{breakDispatchLoop: true, breakRetryLoop: true}

// check
for {
  result := tr.dispatchSingleTaskFromBuffer(isolationGroup, taskInfo)
  if result.breakRetryLoop {
    result.breakDispatchLoop
  }
}
```
^ more verbose obviously, but also pretty hard to screw up vs `return true, false` and relying on correct (but not required) var naming patterns.

Reply:
I don't feel super strongly, but I'm not super sure it's much safer tbh. The defaulting behaviour is not particularly different and Imho unless there's a way to differentiate undefined values from false, it ends up with the same net effect

Reply:
yea, I mean more that relying on IDE variable name autocomplete (particularly in the age of LLMs) for review-correctness seems risky :|

not a big thing, obviously.  just pondering.